DataFormats/SiStripCluster/src/classes_def.xml

```diff
@@ -1,12 +1,19 @@
 <lcgdict>
 
- <class name="SiStripCluster" ClassVersion="13">
+ <class name="SiStripCluster" ClassVersion="14">
+  <version ClassVersion="14" checksum="1374720584"/>
   <version ClassVersion="13" checksum="1374720584"/>
   <version ClassVersion="12" checksum="2984011925"/>
   <version ClassVersion="11" checksum="3702468681"/>
   <version ClassVersion="10" checksum="3791198690"/>
   <field name="error_x" transient="true"/>
  </class>
+ <ioread sourceClass="SiStripCluster" version="[10]" targetClass="SiStripCluster" source="" target="" >
+   <![CDATA[newObj->initQB();]]>
+ </ioread>
+ <ioread sourceClass="SiStripCluster" version="[11-13]" targetClass="SiStripCluster" source="float barycenter_;" target="firstStrip_" >
+   <![CDATA[if (onfile.barycenter_ == 0) { newObj->initQB(); } else { firstStrip_ |= SiStripCluster::approximateMask; }]]>
+ </ioread>
```

Comment:
@cms-sw/core-l2 @pcanal 
this still crashes while reading old data at least in FWLite (it runs OK reading new data).
With some debugging, the crash happens before the ioread rule is executed. The crash is somewhat random at the end point varying between `malloc` complaining of the size `invalid next size (unsorted)` and SIGSEGV. 
Somewhat often in the stack is `int TStreamerInfoActions::ReadSTL`.
Unless something looks obviously wrong in the class_def.xml updates here, 
What's a good way to debug?

Reply:
I think the `target` should include `charge_` and `barycenter_` as well, because the `initQB()` writes to those, and `source` should include `amplitudes_` because `initQB()` reads from those (but I don't know if this would cause the crashes). I also wonder if the read rule for version 10 should also include the `source` and `target` corresponding to what `initQB()` does.

Does cmsRun crash as well? (the test failures with earlier version would seem to indicate so, maybe we should rerun to see) This detail is important because full framework and FWLite handle the construction of read-in objects differently.

Reply:
oh, I see, the schema rule code passes the source and target parts for some pre-processing before calling the actual read rule. 
it's not just a convenient part to access data in the compiled read rule.

I thought that available members are populated first before the read rule is called. 
Apparently nothing is actually filled and everything mentioned in either source or target is skipped:
- after I added all touched data members both in the source and target and called only `initQB` the crash went away but the target appeared empty
- after adding source to target copies explicitly, it looks like the result is OK (at least with a quick fwlite test)

Reply:
@makortel 
thanks for the suggestive direction to make updates.
I hope the last update covers everything, although this looks a bit too verbose (but so be it, as long as it works)

Reply:
> I thought that available members are populated first before the read rule is called.
Apparently nothing is actually filled and everything mentioned in either source or target is skipped:

Indeed, you can not assume anything about newObj (except that it was properly allocated and a constructor was called).
Only the inputs are guaranteed to be from the file (and accessible via the `onfile` handle).